stream/addresses_without_street.js

```diff
@@ -0,0 +1,52 @@
+
+/**
+  This stream is responsible for filling 'addr:street' if is empty or missing,
+  with 'addr:place' value. Note that both 'addr:place' and 'addr:housenumber'
+  must be present inside tags!
+
+  @ref https://github.com/pelias/openstreetmap/pull/565
+**/
+
+const _ = require('lodash');
+const through = require('through2');
+const peliasLogger = require('pelias-logger').get('openstreetmap');
+
+
+module.exports = function(){
+
+  var stream = through.obj( function( doc, enc, next ) {
+
+    try {
+
+      // skip records with no tags
+      var tags = doc.getMeta('tags');
+      if( !tags ){
+        return next( null, doc );
+      }
+
+      // fill addr:street with addr:place if missing
+      var street = tags['addr:street'] || '';
+      if (
+        street === '' &&  // override also if street tag is empty
+        tags.hasOwnProperty('addr:place') &&
+        tags.hasOwnProperty('addr:housenumber')
+      ) {
+        tags['addr:street'] = tags['addr:place'];
+      }
```

Comment:
This is really great, I just wonder if could just be a bit easier to read 🤔 

we could start with the house number requirement, if this isn't set then it's a no-op anyway:

```
// housenumber is required
if (!_.has(tags, 'addr:housenumber')){ return next( null, doc ); }
```

and then *both* strings should be checked for existence and length:

```
const street = _.get(tags, 'addr:street', '').trim();
const place = _.get(tags, 'addr:place', '').trim();

// when street is unset but place is set, use place for the street name
if( _.isEmpty(street) && !_.isEmpty(place) ) {
 _.set(tags, 'addr:street', place);
}
```